File: apps/error-tracking-upload-source-maps/ios/SourceMapExample/SourceMapExampleApp.swift
import Foundation
import PostHog
import SwiftUI

@main
struct SourceMapExampleApp: App {
    init() {
        let environment = ProcessInfo.processInfo.environment
        let config = PostHogConfig(
            projectToken: environment["POSTHOG_PROJECT_TOKEN"] ?? "test_project_token",
            host: environment["POSTHOG_HOST"] ?? "https://us.i.posthog.com"
        )
        PostHogSDK.shared.setup(config)
    }

    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

File: apps/error-tracking-upload-source-maps/android/app/src/main/java/com/example/sourcemap/SourceMapExampleApp.kt
package com.example.sourcemap

import android.app.Application
import com.posthog.android.PostHogAndroid
import com.posthog.android.PostHogAndroidConfig

class SourceMapExampleApp : Application() {
    override fun onCreate() {
        super.onCreate()

        val projectToken = System.getenv("POSTHOG_PROJECT_TOKEN") ?: "test_project_token"
        val host = System.getenv("POSTHOG_HOST") ?: "https://us.i.posthog.com"
        val config = PostHogAndroidConfig(projectToken, host).apply {
            errorTrackingConfig.autoCapture = true
        }
        PostHogAndroid.setup(this, config)
    }
}
